src/main/java/org/truffleruby/core/kernel/KernelNodes.java

```diff
@@ -1356,15 +1356,18 @@ private boolean callerIs(String caller) {
     }
 
     @CoreMethod(names = "require_relative", isModuleFunction = true, required = 1)
-    public abstract static class RequireRelativeNode extends CoreMethodArrayArgumentsNode {
+    @NodeChild(type = RubyNode.class, value = "feature")
+    public abstract static class RequireRelativeNode extends CoreMethodNode {
 
-        @Specialization(guards = "isRubyString(feature)")
-        public boolean requireRelative(DynamicObject feature,
-                @Cached("create()") RequireNode requireNode) {
-            final String featureString = StringOperations.getString(feature);
-            final String featurePath = getFullPath(featureString);
+        @CreateCast("feature")
+        public RubyNode coerceToString(RubyNode feature) {
+            return NameToJavaStringNodeGen.create(feature);
```

Comment:
NameToJavaStringNode accepts Symbol but `require_relative` does not:
```ruby
> require_relative :foo
TypeError: no implicit conversion of Symbol into String
```

But this is fine since the next commit checks with a ToPathNode fist